LTP: Enable s390x on o3, move bare metal setup to main_common.pm

pevik opened the pull request:
Loading load_baremetal_tests() for LTP_BAREMETAL && INSTALL_LTP
was moved to main_ltp.pm which enables this fix also for openSUSE.
Although IPMI is not yet used in o3, be prepared for future.

This required to move also load_baremetal_tests(), which in turn causes
main_ltp.pm to load some modules from main_common.pm and
products/sle/main.pm to load load_baremetal_tests().

Fixes: d83486058 ("LTP: Introduce bare metal tests")

Verification run:
- opensuse-Tumbleweed-DVD-s390x https://openqa.opensuse.org/tests/1221134
- install_ltp baremetal https://openqa.suse.de/tests/4073665 https://openqa.suse.de/tests/4073666
- various QEMU intel based regression tests tests (install_ltp and actual tests) running locally (SLES, QAM, Tumbleweed; both using package and compiling from git): http://quasar.suse.cz/tests/5011, http://quasar.suse.cz/tests/5012, http://quasar.suse.cz/tests/5013, http://quasar.suse.cz/tests/5014, http://quasar.suse.cz/t5016, http://quasar.suse.cz/t5023
http://quasar.suse.cz/t5017, http://quasar.suse.cz/t5018, http://quasar.suse.cz/tests/5021

pevik commented:
@richiejp @mdoucha FYI. For sure it'd be quicker just to copy paste the fix to `products/opensuse/main.pm`:
```
if (get_var('LTP_BAREMETAL') && get_var('INSTALL_LTP')) {
     load_baremetal_tests();
}
```
but I want to add also o3 related fix and don't want to keep copy pasting.

mdoucha commented on lib/main_ltp.pm:
```diff
@@ -113,91 +133,96 @@ sub loadtest_from_runtest_file {
+        load_bootloader_s390x();
+    }
+
+    loadtest_kernel "../installation/bootloader" if is_pvm;
```
Use regular `loadtest()` here?

pevik replied:
Well, yep I can remove custom function and add `"tests/kernel/"` for regular `loadtest()`. I don't have strong preference, both ways looks ok to me.

mdoucha replied:
No, I didn't want you to remove `loadtest_kernel()`. This specific line should just be `loadtest "installation/bootloader" if is_pvm;` (using the `loadtest()` from main_common.pm).

mdoucha replied:
But never mind, since there's no reason to refactor `loadtest()` in this PR now.

pevik replied:
Right :)

frankenmichl commented on lib/main_ltp.pm:
```diff
@@ -18,31 +18,51 @@
     autotest::loadtest("tests/kernel/$test.pm", %args);
 }
 
+sub load_baremetal_tests {
```
I wonder why you would want this in main_ltp.pm - this is nothing ltp related?

pevik replied:
OK, should I move it into `main_common.pm`? The point is to move it from `products/sle/main.pm` to be reachable also by `products/opensuse/main.pm`. LTP tests (and maybe other kernel tests) use the same logic on o3 and osd. Sharing code removes duplicity.

frankenmichl replied:
main_common would be a more reasonable place. I added this for sle only, as we do not have the possibility to use our hardware also for openSUSE.

pevik replied:
I'll put it into main_common, but the this part is code used for any test which requires installation via yast (we use it for sonic & tails, but it's not specific to it):

```
load_boot_tests();                                              
get_var("AUTOYAST") ? load_ayinst_tests() : load_inst_tests();  
load_reboot_tests();
```
And when we dig into the details, also "baremetal" tests name can be confusing because we used it for baremetal tests in kernel & network (which aren't the only bare metal tests).

pevik commented:
Thanks for your feedback. I will:
1. remove `loadtest_kernel()` and use regular `loadtest()`.
2. move `load_baremetal_tests()` to `main_common.pm`. This function can be used for something else than kernel qa baremetal tests, but not sure whether to rename it (what would be a better name).

pevik commented:
Implemented new version, where I moved `load_baremetal_tests()` to `main_common.pm` (2).
Due this change it wasn't needed to touch `loadtest()` (1), so I kept the original.
I also renamed function to `load_kernel_baremetal_tests()`.
Restarting all the tests.

pevik commented:
I added one documentation commit (no actual code change) to get Travis run on different head. That fixed Travis integration in Github.

pevik commented:
Thanks all for review and sorry for impatience (just today I realize that broken Travis prevents merging PR).
